crates/astria-bridge-contracts/src/lib.rs

```diff
@@ -353,47 +353,61 @@ where
         let (ics20_logs, sequencer_logs) =
             futures::future::try_join(get_ics20_logs, get_sequencer_logs)
                 .await
-                .map_err(GetWithdrawalActionsError::get_logs)?;
+                .map_err(GetWithdrawalLogsError::get_logs)?;
 
         // XXX: The calls to `log_to_*_action` rely on only be called if `GetWithdrawalActions`
         // is configured for either ics20 or sequencer withdrawals (or both). They would panic
         // otherwise.
-        ics20_logs
+        Ok(ics20_logs
             .into_iter()
             .map(|log| self.log_to_ics20_withdrawal_action(log))
             .chain(
                 sequencer_logs
                     .into_iter()
                     .map(|log| self.log_to_sequencer_withdrawal_action(log)),
             )
-            .collect()
+            .collect())
     }
 
-    fn log_to_ics20_withdrawal_action(
+    /// Converts a rollup-side smart contract event log to a sequencer-side ics20 withdrawal action.
+    ///
+    /// # Errors
+    /// - If the log does not contain a block number.
+    /// - If the log does not contain a transaction hash.
+    /// - If the log cannot be decoded as a sequencer withdrawal event.
+    /// - If the log does not contain a `recipient` field.
+    /// - If the memo cannot be encoded to json.
+    /// - If calculating the amount using the asset withdrawal divisor overflows.
+    ///
+    /// # Panics
+    /// - If getting a withdrawal event type that it was not configured for. For example, if the
+    ///   configuration was specified to only get sequencer withdrawal events but an ICS20
+    ///   withdrawal is received.
+    pub fn log_to_ics20_withdrawal_action(
```

Comment:
I don't see why this needs to be a pub method? I don't think it's used anywhere outside this lib? You can then also remove the doc comment.